### ISSUE TITLE
Facebook section `Custom Insights` is missing fields

### ISSUE THREAD

**Issue by travjenkins** [estuary/ui#389]
**Bug**
There are three arrays in the Custom Insight section of the Facebook connector that are not displaying correctly.

**Expected**
Fields should render

**Screenshots**
![image](https://user-images.githubusercontent.com/270078/204571433-7f6422cd-be20-49e8-a812-c44dff1d9591.png)

**Desktop (please complete the following information):**
n/a

**Additional context**
I believe this is due to how these arrays are nested within another array. If you move the `enum` out of the `items` parent the fields render.

**Comment by jshearer** [comment]
So, it took me a little while to dig through the various layers of abstraction here, but the first error here is talking about the field `custom_insights.0.action_breakdowns`, whose schema is defined as:

```json
"action_breakdowns": {
  "default": [],
  "description": "A list of chosen action_breakdowns for action_breakdowns",
  "items": {
    "description": "Generic enumeration.\n\nDerive from this class to define new enumerations.",
    "enum": [
      "action_canvas_component_name",
      "action_carousel_card_id",
      "action_carousel_card_name",
      "action_destination",
      "action_device",
      "action_reaction",
      "action_target_id",
      "action_type",
      "action_video_sound",
      "action_video_type"
    ],
    "title": "ValidActionBreakdowns"
  },
  "title": "Action Breakdowns",
  "type": "array"
}
```

Critically, the inner enum is missing a `"type": "string"` property! Defining the type for this, and the other two enums (`custom_insights.0.breakdowns.0`, and `custom_insights.0.fields.0`) shows the following form:

<img width="1482" alt="Screen Shot 2022-12-12 at 20 26 34" src="https://user-images.githubusercontent.com/4368270/207203459-8a0572ef-2e09-4889-ad34-0c64880b6b1c.png">
<img width="1490" alt="Screen Shot 2022-12-12 at 20 26 41" src="https://user-images.githubusercontent.com/4368270/207203465-32c6cae7-9877-4840-a09c-2006ce59b587.png">

There are still a bunch of missing renderers, but after seeing the above, I strongly suspect that the problem is just that **the schema is incomplete**. I don't really want to just guess what the schema should be, so does someone from the connectors team want to work with me to fix this schema? Maybe @mdibaiee?

**Comment by jshearer** [comment]
A little hack to jsonforms shows this useful debugging output:

<img width="776" alt="Screen Shot 2022-12-12 at 21 00 32" src="https://user-images.githubusercontent.com/4368270/207207730-ab68916c-aff5-44be-8c2f-20d1d643c4aa.png">

**Comment by jshearer** [comment]
Something is very strange.

The schema for `custom_insights` as defined in the `connector_tags.endpoint_spec_schema` column, according to the supabase web UI, contains the following fields:

* `action_breakdowns`
* `breakdowns`
* `end_date`
* `fields`
* `insights_lookback_window`
* `name`
* `start_date`
* `time_increment`

But, when requesting the schema from supabase's API using this query:

```
GET | http://localhost:5431/rest/v1/connector_tags?select=connectors(image_name),id,connector_id,image_tag,endpoint_spec_schema,resource_spec_schema,documentation_url&or=(id.eq.06:8e:ce:62:1a:6a:38:00,connector_id.eq.06:8e:ce:62:1a:6a:38:00)
```

The resulting schema document's `custom_insights` has all of the above fields... and _also_ the following fields:
* ```json
  "max_batch_size": {
    "description": "Maximum batch size used when sending batch requests to Facebook API. You won't need to set this field for most use cases."
  }
  ```
* ```json
  "page_size": {
    "description": "Page size used when sending requests to Facebook API to specify number of records per page when response has pagination. You won't need to set this field for most use cases."
  }
  ```

**Comment by jshearer** [comment]
OK! The `nulls` I was seeing above was actually a false alarm: somehow some invalid fields had gotten into my local database, which aren't present in production. This _did_ present an opportunity to add better handling for these scenarios, which I did. After removing the offending fields from my schema, I see this, which is what we expect:
<img width="511" alt="Screen Shot 2022-12-12 at 22 25 47" src="https://user-images.githubusercontent.com/4368270/207219169-09e411b6-6425-4f4a-b8c5-bcf9cceac524.png">

I went ahead and made the schema changes on prod, which amounted to adding `"type": "string"` to `custom_insights.0.action_breakdowns.items`, `custom_insights.0.breakdowns.items`, and `custom_insights.0.fields.items`. 

As of posting this, I can confirm that prod no longer shows `No applicable renderer found`!
